test/DisputeManager.test.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import {expect} from 'chai';
+import {ethers, waffle} from 'hardhat';
+import {deployContracts} from './setup';
+import {etherParse, registerIndexer, startNewEra, time} from './helper';
+import {DEPLOYMENT_ID} from './constants';
+import {DisputeManager, SQToken, Staking, IndexerRegistry, EraManager, RewardsDistributer, RewardsStaking, RewardsHelper} from '../src';
+import {utils} from 'ethers';
+
+describe('Dispute Manager Contract', () => {
+    const mockProvider = waffle.provider;
+    let root, indexer, fisherman;
+    let disputeManager: DisputeManager;
+    let token: SQToken;
+    let staking: Staking;
+    let indexerRegistry: IndexerRegistry;
+    let eraManager: EraManager;
+    let rewardsDistributor: RewardsDistributer;
+    let rewardsStaking: RewardsStaking;
+    let rewardsHelper: RewardsHelper;
+
+    beforeEach(async () => {
+        [root, indexer, fisherman] = await ethers.getSigners();
+        const deployment = await deployContracts(root, root);
+        disputeManager = deployment.disputeManager;
+        token = deployment.token;
+        staking = deployment.staking;
+        indexerRegistry = deployment.indexerRegistry;
+        eraManager = deployment.eraManager;
+        rewardsDistributor = deployment.rewardsDistributer;
+        rewardsStaking = deployment.rewardsStaking;
+        rewardsHelper = deployment.rewardsHelper;
+        await eraManager.updateEraPeriod(time.duration.days(5).toString());
+        await token.connect(fisherman).increaseAllowance(disputeManager.address, etherParse("1000"));
+        await registerIndexer(token, indexerRegistry, staking, root, indexer, '2000');
+    });
+
+    it('set MinimumDeposit should work', async () => {
+        expect(await disputeManager.minimumDeposit()).to.equal(etherParse('1000'));
+        await disputeManager.setMinimumDeposit(etherParse('10'));
+        expect(await disputeManager.minimumDeposit()).to.equal(etherParse('10'));
+    });
+
+    it('createDispute should work', async () => {
+        await token.connect(root).transfer(fisherman.address, etherParse("1000"));
+        let tx = await disputeManager.connect(fisherman).createDispute(indexer.address, DEPLOYMENT_ID, etherParse('1000'), 0);
+        let receipt = await tx.wait();
+        let evt = receipt.events.find(
```

Comment:
[suggestion] build a helper function to check the event, this can also use in all the places we check event result

```
async function txEvent(tx: Tx, event: string): Event {
 ....
}
```

Reply:
since also need to pass the instance of the contract to it to get event:
```
const event = disputeManager.interface.decodeEventLog(
                disputeManager.interface.getEvent('DisputeOpen'),
                evt.data
);
```
I dont think we can make it...

Reply:
Try this:

```
type Event = utils.Result;
async function eventFrom(tx: ContractTransaction, contract: BaseContract, event: string): Promise<Event> {
    const receipt = await tx.wait();
    const evt = receipt.events.find((log) => log.topics[0] === utils.id(event));

    const eventName = event.split('(')[0];
    return contract.interface.decodeEventLog(
        contract.interface.getEvent(eventName),
        evt.data
    );
}
```